src/components/SamlConfiguration/index.test.jsx

```diff
@@ -1,47 +1,78 @@
 import React from 'react';
-import { shallow } from 'enzyme';
+import { shallow, mount } from 'enzyme';
 
 import SamlConfiguration from './index';
 import LmsApiService from './../../data/services/LmsApiService';
 import { REQUIRED_DATA_FIELDS } from './SamlProviderDataForm';
+import { REQUIRED_CONFIG_FIELDS } from './SamlProviderConfigForm';
+
+jest.mock('./../../data/services/LmsApiService');
 
 const formData = new FormData();
-const responseData = { data: {} };
+const configResponse = {
+  data: { results: [{}] },
+};
+const validDataResponse = {
+  data: { results: [{}] },
+};
+const invalidDataResponse = {
+  data: { results: [{}] },
+};
+
 const oldConfig = { };
-REQUIRED_DATA_FIELDS.forEach((field) => {
+REQUIRED_CONFIG_FIELDS.forEach((field) => {
   formData.append(field, 'testdata');
-  responseData.data[field] = 'testdata';
+  configResponse.data.results[0][field] = 'testdata';
   oldConfig[field] = 'oldtestdata';
 });
+REQUIRED_DATA_FIELDS.forEach((field) => {
+  validDataResponse.data.results[0][field] = 'testdata';
+});
 
 describe('<SamlConfiguration /> ', () => {
+  it('get provider config (with no data)', () => {
+    LmsApiService.getProviderConfig.mockResolvedValue(configResponse);
+    LmsApiService.getProviderData.mockResolvedValue(invalidDataResponse);
+    const wrapper = mount(<SamlConfiguration enterpriseId="testEnterpriseId" />);
+    setImmediate(() => {
+      expect(wrapper.state().providerConfig).toEqual(configResponse.data.results[0]);
+    });
+  });
+
+  it('get provider config and data', () => {
+    LmsApiService.getProviderConfig.mockResolvedValue(configResponse);
+    LmsApiService.getProviderData.mockResolvedValue(validDataResponse);
+    const wrapper = mount(<SamlConfiguration enterpriseId="testEnterpriseId" />);
+    setImmediate(() => {
+      expect(wrapper.state().providerConfig).toEqual(configResponse.data.results[0]);
+      expect(wrapper.state().providerData).toEqual(validDataResponse.data.results[0]);
+    });
+  });
+
   it('creating provider config states', () => {
     const wrapper = shallow(<SamlConfiguration enterpriseId="testEnterpriseId" />);
-    const spyPostNewProviderConfig = jest.spyOn(LmsApiService, 'postNewProviderConfig');
 
-    spyPostNewProviderConfig.mockImplementation(() => responseData);
+    LmsApiService.postNewProviderConfig.mockResolvedValue(configResponse);
```

Comment:
oh yeah, I forgot `mock resolved value`. 👍

Reply:
I previously did not know about it, but it makes sense.